.github/workflows/rust.yml

```diff
@@ -88,10 +88,16 @@ jobs:
       - name: Checkout sources
         uses: actions/checkout@v4
       - uses: Swatinem/rust-cache@v2
-      - name: Run cargo fmt
+      - name: Install yamlfmt
+        env:
+          YFV: "0.13.0"
+          HASH: "043e96d754a8afa4f4c5c13ffb2f3e50c6be5a70bf53292d3025abc0b42fe4ae"
         run: |
-          rustup component add rustfmt
-          cargo fmt --all -- --check
+          curl -L --fail --output /tmp/yamlfmt.tar.xz https://github.com/google/yamlfmt/releases/download/v${YFV}/yamlfmt_${YFV}_Linux_x86_64.tar.gz
```

Comment:
nit: If we would add `nix`, this could be a simple `nix-shell --run "cargo xtask fmt --check"`

Reply:
Makes sense, but looks like a bit of additional work is needed since the nix files currently fail the format check, and also https://github.com/nix-community/nixpkgs-fmt has been deprecated. I'll go ahead and merge the version without nix, and we can follow up with another PR later.

Reply:
Ohh.. I had no clue that it is deprecated (to be fair, only since very recently) . Thanks!